frontend/src/app/components/containers/complaints/outcomes/ceeb/ceeb-decision/decision-form.tsx

```diff
@@ -222,29 +225,100 @@ export const DecisionForm: FC<props> = ({
     );
   };
 
+  const resetErrorMessages = () => {
+    setScheduleErrorMessage("");
+    setSectorErrorMessage("");
+    setDischargeErrorMessage("");
+    setActionTakenErrorMessage("");
+    setAssignedToErrorMessage("");
+    setDateActionTakenErrorMessage("");
+    setLeadAgencyErrorMessage("");
+    setInspectionNumberErrorMessage("");
+  };
+
   const handleSaveButtonClick = () => {
     const identifier = id !== undefined ? caseId : leadIdentifier;
+    resetErrorMessages();
 
-    dispatch(upsertDecisionOutcome(identifier, data)).then(async (response) => {
-      if (response === "success") {
-        //-- update the assignment of the complaint to the selected officer
-        const { assignedTo } = data;
-        const result = await dispatch(assignComplaintToOfficer(leadIdentifier, assignedTo));
+    if (isValid()) {
+      dispatch(upsertDecisionOutcome(identifier, data)).then(async (response) => {
+        if (response === "success") {
+          //-- update the assignment of the complaint to the selected officer
+          const { assignedTo } = data;
+          if (assignedTo) {
+            const result = await dispatch(assignComplaintToOfficer(leadIdentifier, assignedTo));
 
-        if (result) {
-          //-- update the complaint
-          dispatch(getComplaintById(leadIdentifier, COMPLAINT_TYPES.ERS));
-        } else {
-          ToggleError("Error, unable to to assign officer to complaint");
-        }
+            if (result) {
+              //-- update the complaint
+              dispatch(getComplaintById(leadIdentifier, COMPLAINT_TYPES.ERS));
+            } else {
+              ToggleError("Error, unable to to assign officer to complaint");
+            }
+          }
 
-        dispatch(getCaseFile(leadIdentifier));
+          dispatch(getCaseFile(leadIdentifier));
 
-        if (id !== undefined) {
-          toggleEdit(false);
+          if (id !== undefined) {
+            toggleEdit(false);
+          }
         }
+      });
+    }
+  };
+
+  const isValid = (): boolean => {
+    let _isValid = true;
+
+    if (!data.schedule) {
+      setScheduleErrorMessage("Required");
+    }
+
+    if (!data.sector) {
+      setSectorErrorMessage("Required");
+    }
+
+    if (!data.discharge) {
+      setDischargeErrorMessage("Required");
+    }
+
+    if (!data.schedule || !data.sector || !data.discharge) {
+      _isValid = false;
```

Comment:
Could cognitive complexity be reduced by 2 here by putting this into the above conditionals?